core/controllers/acl_decorators_test.py

```diff
@@ -5175,6 +5204,38 @@ def test_user_cannot_access_story_when_topic_is_not_published(self) -> None:
                 % self.story_url_fragment,
                 expected_status_int=404)
 
+    def test_guest_cannot_access_story(self) -> None:
+        self.logout()
+        with self.swap(self, 'testapp', self.mock_testapp):
+            response = self.mock_testapp.get(
+                '/mock_story_data/staging/topic/%s' % self.story_url_fragment,
+                expect_errors=True)
+            self.assertEqual(response.status_int, 401)
+
+    def test_user_cannot_access_story_with_empty_topic_id(self) -> None:
+        invalid_story_id = story_services.get_new_story_id()
+        self.save_new_story(
+            # Empty topic_id.
+            invalid_story_id, self.admin_id, '',
+            url_fragment='story-no-topic')
+
+        with self.swap(self, 'testapp', self.mock_testapp):
+            self.get_json(
+                '/mock_story_data/staging/topic/story-no-topic',
+                expected_status_int=404)
+
+    def test_user_cannot_access_story_with_wrong_story_id(self) -> None:
```

Comment:
Can you clarify this a bit more?

The ID of a story is going to be its story ID, by definition. So, what exactly does "wrong" mean in this case?